src/loop_logic.rs

```diff
@@ -477,17 +479,20 @@ impl<'l, Data> EventLoop<'l, Data> {
                             .inner
                             .sources
                             .borrow_mut()
-                            .remove(registroken_token);
+                            .get_mut(registroken_token)
+                            .take();
```

Comment:
Why are you not using `try_remove` here?

Reply:
Because we want to intentionally leave `None` in place, while there are still events being processed. I added some more comments to explain this.

Reply:
Ironically, this is exactly where the issue causing the CI failure is:

```suggestion
                            .get_mut(registroken_token)
                            .unwrap_or(&mut None)
                            .take();
```